enhancements/machine-api/control-plane-machineset-vertical-scaling.md

```diff
@@ -0,0 +1,348 @@
+to allow it to make automated scaling decisions for control plane node sizing.
+
+## Motivation
+
+During normal operation of an OpenShift cluster, as worker node count and
+workloads increase, it becomes necessary to increase resources available to the
+control plane nodes.
+
+During SRE operations of the managed OpenShift product, there was already a
+trigger indentified, when these increases are required. Instead of requiring the
+adjustments to be performed by hand, the operator should be able to
+automatically determine if a control plane node size change is required.
+
+On the other hand, it should also be possible to determine that control plane
+nodes have become too large for a cluster, because load has decreased. In this
+case the operator should also be able to automatically decide to scale the
+control plane down again.
+
+With the automatic increase of the control plane size, cluster performance can
+remain high, while automated decrease of the size will ensure that spend for
+users in a cloud environment is only has high as needed.
+
+### User Stories
+
+* As an OpenShift administrator, I want to know that the platform will always
+  have enough resources available to it's control plane nodes without manual
+  intervention.
+* As an OpenShift administrator, I want to know that the platform can reduce
+  spending on control plane nodes if they are oversized for the current cluster
+  load.
+* As an OpenShift administrator, I want to be able to control when sizing up and
+  down occurs, to adjust it to workloads on my cluster that might be burstable
+  to prevent the automatic scaling from occurring to often.
+
+### Goals
+
+* Allow automatic vertical scaling of control plane nodes.
+* Allow users to configure the sizes that can be chosen for automatic scaling.
+* Allow users to configure the thresholds when scaling will occur for CPU and
+  memory usage.
+* Allow users to configure how often checks for automatic scaling should be run.
+* Allow users to configure how long after a scale up or scale down, no more
+  scaling should be performed.
+* Allow users to completely disable scale up.
+* Allow users to completely disable scale down.
+
+### Non-Goals
+
+* Horizontal scaling of control plane nodes
+
+## Proposal
+
+New configuration and a reconcile loop for the
+`control-plane-machine-set-operator` will be introduced that will monitor
+resource usage of the control plane nodes and automatically modify the
+`ControlPlaneMachineSet` if scaling the nodes is determined to be required.
+
+The new configuration will specify if scaling up and down is enabled and allow
+configuring when to trigger scaling. It will also allow configuring which
+machine types are to be used for scaling, putting an upper and lower limit on
+possible automatic scaling.
+
+### Workflow Description
+
+1. The OpenShift administrator installs the
+   `control-plane-machine-set-operator`.
+2. The OpenShift adminstrator creates a valid `control-plane-machine-set`
```

Comment:
Configmaps are not APIs, any configuration that is needed should be via a concrete kube API. Whether that's by adding new config to the CPMS API or a new custom resource, it should be something structured